prog/probe.go

```diff
@@ -221,3 +233,28 @@ func getFirstAddressOf(name string) (string, error) {
 		return "", fmt.Errorf("No address found for %s", name)
 	}
 }
+
+// TODO(paulbellamy): where should this live?
+func pluginsReporter(pluginRegistry *plugins.Registry) probe.Reporter {
+	return probe.ReporterFunc("plugins", func() (report.Report, error) {
+		rpt := report.MakeReport()
+		pluginRegistry.Implementors("reporter", func(plugin *plugins.Plugin) {
+			pluginReportJSON := map[string]interface{}{}
+			if err := plugin.Call("Report", nil, &pluginReportJSON); err != nil {
+				log.Errorf("plugins: error getting report from %s: %v", plugin.ID, err)
+			}
+
+			pluginReport := report.MakeReport()
+			pluginReport.Plugins = pluginReport.Plugins.Add(plugin)
+			buf := &bytes.Buffer{}
+			if err := codec.NewEncoder(buf, &codec.JsonHandle{}).Encode(pluginReportJSON); err != nil {
+				log.Errorf("plugins: error pointlessly encoding/decoding report: %v", err)
+			}
+			if err := codec.NewDecoder(buf, &codec.JsonHandle{}).Decode(&pluginReport); err != nil {
+				log.Errorf("plugins: error pointlessly encoding/decoding report: %v", err)
+			}
```

Comment:
Yeah, will figure out a better way around this (that's the "make the code sane" bit on the todo list).